Error: ./configureAndroidSdk.sh exited with non-zero code: 1

### New Issue Checklist

- [x] Updated turtle-cli to the latest version (`yarn global add turtle-cli` or `npm install --global turtle-cli`).
- [x] I read the [Contribution Guidelines](https://github.com/expo/turtle/blob/master/CONTRIBUTING.md).
- [x] I read [docs.expo.io](https://docs.expo.io) (Managed Workflow > Distributing Your App).
- [x] I searched for [existing GitHub issues](https://github.com/expo/turtle/issues).
- [x] **I enclosed the complete output when running turtle, including the stack trace and command used.**

### Issue Description
Error happened on Ubuntu 18.04 with latest everything. Issue also happened on Windows but I understand that's not supported either way the same issue is happening on Linux which the console log was grabbed from.

##### Command executed
turtle setup:android

##### Complete output when running turtle, including the stack trace and command used

<details>
  <pre>
turtle setup:android
Oct 26 06:01:53 turtle[3552] INFO:  Downloading Android SDK
  platform: "android"
  buildPhase: "setting up environment"
downloading [=================== ] 96% 0.6sOct 26 06:02:14 turtle[3552] INFO:  Decompressing Android SDK
  platform: "android"
  buildPhase: "setting up environment"
Oct 26 06:02:16 turtle[3552] INFO:  Configuring Android SDK, this may take a while
  platform: "android"
  buildPhase: "setting up environment"
Oct 26 06:02:17 turtle[3552] INFO:  Exception in thread "main"
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:  java.lang.NoClassDefFoundError: javax/xml/bind/annotation/XmlSchema
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:      at com.android.repository.api.SchemaModule$SchemaModuleVersion.<init>(SchemaModule.java:156)
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:      at com.android.repository.api.SchemaModule.<init>(SchemaModule.java:75)
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:      at com.android.sdklib.repository.AndroidSdkHandler.<clinit>(AndroidSdkHandler.java:81)
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:      at com.android.sdklib.tool.sdkmanager.SdkManagerCli.main(SdkManagerCli.java:73)
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:      at com.android.sdklib.tool.sdkmanager.SdkManagerCli.main(SdkManagerCli.java:48)
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:  Caused by: java.lang.ClassNotFoundException: javax.xml.bind.annotation.XmlSchema
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:      at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:583)
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:      at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:      at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:      ... 5 more
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:  Exception in thread "main"
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:  java.lang.NoClassDefFoundError: javax/xml/bind/annotation/XmlSchema
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:      at com.android.repository.api.SchemaModule$SchemaModuleVersion.<init>(SchemaModule.java:156)
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:      at com.android.repository.api.SchemaModule.<init>(SchemaModule.java:75)
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:      at com.android.sdklib.repository.AndroidSdkHandler.<clinit>(AndroidSdkHandler.java:81)
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:      at com.android.sdklib.tool.sdkmanager.SdkManagerCli.main(SdkManagerCli.java:73)
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:      at com.android.sdklib.tool.sdkmanager.SdkManagerCli.main(SdkManagerCli.java:48)
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:  Caused by: java.lang.ClassNotFoundException: javax.xml.bind.annotation.XmlSchema
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:      at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:583)
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:      at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:      at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:      ... 5 more
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:  Exception in thread "main"
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:  java.lang.NoClassDefFoundError: javax/xml/bind/annotation/XmlSchema
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:      at com.android.repository.api.SchemaModule$SchemaModuleVersion.<init>(SchemaModule.java:156)
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:      at com.android.repository.api.SchemaModule.<init>(SchemaModule.java:75)
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:      at com.android.sdklib.repository.AndroidSdkHandler.<clinit>(AndroidSdkHandler.java:81)
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:      at com.android.sdklib.tool.sdkmanager.SdkManagerCli.main(SdkManagerCli.java:73)
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:      at com.android.sdklib.tool.sdkmanager.SdkManagerCli.main(SdkManagerCli.java:48)
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:  Caused by: java.lang.ClassNotFoundException: javax.xml.bind.annotation.XmlSchema
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:      at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:583)
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:      at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:      at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:      ... 5 more
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:  Exception in thread "main"
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:  java.lang.NoClassDefFoundError: javax/xml/bind/annotation/XmlSchema
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:      at com.android.repository.api.SchemaModule$SchemaModuleVersion.<init>(SchemaModule.java:156)
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:      at com.android.repository.api.SchemaModule.<init>(SchemaModule.java:75)
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:      at com.android.sdklib.repository.AndroidSdkHandler.<clinit>(AndroidSdkHandler.java:81)
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:      at com.android.sdklib.tool.sdkmanager.SdkManagerCli.main(SdkManagerCli.java:73)
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:      at com.android.sdklib.tool.sdkmanager.SdkManagerCli.main(SdkManagerCli.java:48)
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:  Caused by: java.lang.ClassNotFoundException: javax.xml.bind.annotation.XmlSchema
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:      at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:583)
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:      at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:      at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:17 turtle[3552] INFO:      ... 5 more
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:18 turtle[3552] INFO:  Exception in thread "main"
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:18 turtle[3552] INFO:  java.lang.NoClassDefFoundError: javax/xml/bind/annotation/XmlSchema
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:18 turtle[3552] INFO:      at com.android.repository.api.SchemaModule$SchemaModuleVersion.<init>(SchemaModule.java:156)
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:18 turtle[3552] INFO:      at com.android.repository.api.SchemaModule.<init>(SchemaModule.java:75)
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:18 turtle[3552] INFO:      at com.android.sdklib.repository.AndroidSdkHandler.<clinit>(AndroidSdkHandler.java:81)
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:18 turtle[3552] INFO:      at com.android.sdklib.tool.sdkmanager.SdkManagerCli.main(SdkManagerCli.java:73)
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:18 turtle[3552] INFO:      at com.android.sdklib.tool.sdkmanager.SdkManagerCli.main(SdkManagerCli.java:48)
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:18 turtle[3552] INFO:  Caused by: java.lang.ClassNotFoundException: javax.xml.bind.annotation.XmlSchema
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:18 turtle[3552] INFO:      at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:583)
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:18 turtle[3552] INFO:      at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:18 turtle[3552] INFO:      at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:18 turtle[3552] INFO:      ... 5 more
  platform: "android"
  buildPhase: "setting up environment"
  source: "stderr"
Oct 26 06:02:18 turtle[3552] ERROR: Failed to setup environment for android builds
  err: Error: ./configureAndroidSdk.sh exited with non-zero code: 1
      at ChildProcess.completionListener (/usr/local/lib/node_modules/turtle-cli/node_modules/@expo/spawn-async/src/spawnAsync.ts:65:13)
      at Object.onceWrapper (events.js:317:30)
      at emitTwo (events.js:126:13)
      at ChildProcess.emit (events.js:214:7)
      at maybeClose (internal/child_process.js:925:16)
      at Process.ChildProcess._handle.onexit (internal/child_process.js:209:5)
      ...
      at spawnAsync (/usr/local/lib/node_modules/turtle-cli/node_modules/@expo/spawn-async/src/spawnAsync.ts:26:19)
      at Object.spawnAsyncThrowError (/@expo/xdl@56.2.2-alpha.2/src/detach/ExponentTools.ts:131:19)
      at _configureSdk (/usr/local/lib/node_modules/turtle-cli/src/bin/setup/android/sdk.ts:54:23)
      at <anonymous>
  platform: "android" 
 </pre>
</details>

### Environment

<!-- Please run `expo diagnostics` and copy the output below. This will help us help you. -->

<details>
  <pre>
  Expo CLI 3.4.1 environment info:
    System:
      OS: Linux 4.15 Ubuntu 18.04.3 LTS (Bionic Beaver)
      Shell: 4.4.20 - /bin/bash
    Binaries:
      Node: 8.10.0 - /usr/bin/node
      Yarn: 1.19.1 - /usr/local/bin/yarn
      npm: 3.5.2 - /usr/bin/npm
    npmPackages:
      expo: ^35.0.0 => 35.0.0
      react: 16.8.3 => 16.8.3
      react-native: https://github.com/expo/react-native/archive/sdk-35.0.0.tar.gz => 0.59.8
      react-navigation: ^4.0.10 => 4.0.10
    npmGlobalPackages:
      expo-cli: 3.4.1</pre>
</details>

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,6 +7,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## Unreleased
 
+### Changed
+- Enforce Java 8 when using turtle-cli.
+
 ## [0.12.7] - 2019-10-25
 
 ### Changed

--- a/src/bin/setup/android/index.ts
+++ b/src/bin/setup/android/index.ts
@@ -15,6 +15,7 @@ import { PLATFORMS } from 'turtle/constants';
 import logger from 'turtle/logger';
 
 const which = util.promisify(_which);
+const JAVA_REQUIRED_VERSION = 8;
 const REQUIRED_TOOLS: IToolDefinition[] = [
   {
     command: 'bash',
@@ -22,16 +23,27 @@ const REQUIRED_TOOLS: IToolDefinition[] = [
   },
   {
     command: 'javac',
-    missingDescription: 'Please install JDK (version 8 or newer) - check https://jdk.java.net/',
+    missingDescription: `Please install JDK (version ${JAVA_REQUIRED_VERSION}) - check https://jdk.java.net/`,
     testFn: async () => {
-      const { status, stdout } = await ExponentTools.spawnAsyncThrowError(
-        'javac',
+      const { status, stdout, stderr } = await ExponentTools.spawnAsyncThrowError(
+        'java',
         ['-version'],
         { stdio: 'pipe' },
       );
       if (stdout.match(/no java runtime present/i)) {
         return false;
       }
+
+      const matchResult = stderr.match(/.*version ".*\.(.*)\..*"/);
+      if (matchResult) {
+        const [, currentJavaVersion] = matchResult;
+        if (Number(currentJavaVersion) !== JAVA_REQUIRED_VERSION) {
+          throw new Error(`You're using a wrong Java version, please install version ${JAVA_REQUIRED_VERSION}`);
+        }
+      } else {
+        logger.warn(`Couldn't find Java version number, assuming you're on Java ${JAVA_REQUIRED_VERSION}...`);
+      }
+
       return status === 0;
     },
   },

